arcgis-ios-sdk-samples/Layers/Query with CQL filters/QueryWithCQLFiltersSettingsViewController.swift

```diff
@@ -0,0 +1,176 @@
+}
+
+class QueryWithCQLFiltersSettingsViewController: UITableViewController {
+    // MARK: Storyboard views
+    
+    /// The cell for where clause options.
+    @IBOutlet var whereClauseCell: UITableViewCell!
+    /// The cell for max features input.
+    @IBOutlet var maxFeaturesCell: UITableViewCell! {
+        didSet {
+            maxFeaturesCell.detailTextLabel?.text = String(maxFeatures)
+        }
+    }
+    /// The start date picker.
+    @IBOutlet var startDatePicker: UIDatePicker!
+    /// The end date picker.
+    @IBOutlet var endDatePicker: UIDatePicker!
+    /// The switch to control if date pickers are enabled.
+    @IBOutlet var dateFilterSwitch: UISwitch!
+    
+    // MARK: Properties
+    
+    /// A delegate to notify other view controllers.
+    weak var delegate: QueryWithCQLFiltersSettingsViewControllerDelegate?
+    
+    let numberFormatter = NumberFormatter()
+    let sampleWhereClauses = [
+        // Sample Query 1: Features with an F_CODE property of "AP010".
+        // A cql-text query.
+        "F_CODE = 'AP010'",
+        // A cql-json query.
+        "{ \"eq\" : [ { \"property\" : \"F_CODE\" }, \"AP010\" ] }",
+        // Sample Query 2: Features with an F_CODE property matching the pattern.
+        "F_CODE LIKE 'AQ%'",
+        // Sample Query 3: Use cql-json to combine "before" and "eq" operators
+        // with the logical "and" operator.
+        "{\"and\":[{\"eq\":[{ \"property\" : \"F_CODE\" }, \"AP010\"]},{ \"before\":[{ \"property\" : \"ZI001_SDV\"},\"2013-01-01\"]}]}"
+    ]
+    var selectedWhereClause: String?
+    var maxFeatures = 1000
+    
+    // MARK: Methods and Actions
+    
+    func makeQueryParameters() -> AGSQueryParameters {
+        let queryParameters = AGSQueryParameters()
+        queryParameters.whereClause = selectedWhereClause ?? ""
+        queryParameters.maxFeatures = maxFeatures
+        if dateFilterSwitch.isOn {
+            queryParameters.timeExtent = AGSTimeExtent(startTime: startDatePicker.date, endTime: endDatePicker.date)
+        }
+        return queryParameters
+    }
+    
+    func showWhereClausePicker() {
+        let selectedIndex = sampleWhereClauses.firstIndex { $0 == selectedWhereClause }
+        let optionsViewController = OptionsTableViewController(labels: sampleWhereClauses, selectedIndex: selectedIndex) { newIndex in
+            let whereClause = self.sampleWhereClauses[newIndex]
+            self.selectedWhereClause = whereClause
+            self.whereClauseCell.detailTextLabel?.text = whereClause
+            self.navigationController?.popViewController(animated: true)
+        }
+        optionsViewController.title = "Where Clause"
+        show(optionsViewController, sender: self)
+    }
+    
+    /// Prompt an alert to allow the user to input a numeric value.
+    func showValueInputField(completion: @escaping (NSNumber) -> Void) {
```

Comment:
An alert really shouldn't be used for this. Instead, the cell should just have a text field that the user can edit directly. See `StretchRendererSettingsViewController` for example.

Reply:
I've made some changes, please let me know what you think!

- I didn't draw the text field in a custom cell, but added it as the accessory view to a `Basic` cell to avoid the complexity of setting up constraints.
- (as always 🤦 ) I couldn't find an elegant way to dismiss the `.numberPad` keyboard in a table view, so I cheated a bit to dismiss it with scroll. See in the `viewDidLoad` method: `tableView.keyboardDismissMode = .onDrag`
- The text will be silently reset to default value if the input is invalid.